anime_downloader/sites/helpers/selescrape.py

```diff
@@ -0,0 +1,182 @@
+    binary_path = data['dl']['selescrape_driver_binary_path'].lower()
+    return binary_path
+
+
+def add_url_params(url, params):
+    encoded_params = urlencode(params)
+    url = url + '?' + encoded_params
+    return url
+
+
+def driver_select(): #
+    """
+    it configures what each browser should do 
+    and gives the driver variable that is used 
+    to perform any actions below this def
+    """
+    browser = get_browser_config()
+    data_dir = get_data_dir()
+    executable = get_browser_executable()
+    driver_binary = get_driver_binary()
+    if driver_binary == '.':
+        binary = None
+    else:
+        binary = driver_binary
+    if browser == 'firefox':
+        fireFoxOptions = webdriver.FirefoxOptions()
+        fireFoxOptions.headless = True
+        fireFoxOptions.add_argument('--log fatal')
+        if driver_binary == '.':  
+            driver = webdriver.Firefox(options=fireFoxOptions, service_log_path=os.path.devnull)
+        else:
+            try:
+                driver = webdriver.Firefox(options=fireFoxOptions, service_log_path=os.path.devnull)
+            except:
+                driver = webdriver.Firefox(executable_path=driver_binary, options=fireFoxOptions, service_log_path=os.path.devnull)
+    elif browser == 'chrome':
+        from selenium.webdriver.chrome.options import Options
+        chrome_options = Options()
+        chrome_options.add_argument("--headless")
+        chrome_options.add_argument("--disable-gpu")
+        profile_path = os.path.join(data_dir, 'Selenium_chromium')
+        log_path = os.path.join(data_dir, 'chromedriver.log')
+        chrome_options.add_argument(f'--log-path {log_path}')
+        chrome_options.add_argument(f"--user-data-dir={profile_path}")
+        chrome_options.add_argument("--no-sandbox")
+        chrome_options.add_argument("--window-size=1920,1080")
+        chrome_options.add_argument(f'user-agent={get_random_header()}')
+        if driver_binary == '.':
+            if executable == '.':
+                driver = webdriver.Chrome(options=chrome_options)
+            else:
+                from selenium.webdriver.common.desired_capabilities import DesiredCapabilities
+                cap = DesiredCapabilities.CHROME
+                cap['binary_location'] = executable
+                driver = webdriver.Chrome(desired_capabilities=cap, options=chrome_options)
+        else:
+            if executable == '.':
+                driver = webdriver.Chrome(options=chrome_options)
+            else:
+                from selenium.webdriver.common.desired_capabilities import DesiredCapabilities
+                cap = DesiredCapabilities.CHROME
+                cap['binary_location'] = executable
+                driver = webdriver.Chrome(executable_path=driver_binary, desired_capabilities=cap, options=chrome_options)
+    return driver
+
+
+def status_select(driver, url, status='hide'):
+    """
+    for now it doesnt do what its name suggests, 
```

Comment:
:shrug:

Reply:
Literal amazing code design do you work for Valve? :o

Reply:
im dying from this one